src/_pytest/fixtures.py

```diff
@@ -1245,21 +1217,57 @@ def __call__(self, function: FixtureFunction) -> FixtureFunction:
                 pytrace=False,
             )
 
-        # Type ignored because https://github.com/python/mypy/issues/2087.
-        function._pytestfixturefunction = self  # type: ignore[attr-defined]
-        return function
+        return fixture_definition
+
+
+# TODO: paramspec/return type annotation tracking and storing
+class FixtureFunctionDefinition:
+    def __init__(
+        self,
```

Comment:
These days I prefer to force keyword arguments in such cases:

```suggestion
        self,
        *,
```